src/ILVerify/src/Program.cs

```diff
@@ -220,26 +251,56 @@ private IEnumerable<VerificationResult> VerifyAssembly(AssemblyName name, out Ec
             PEReader peReader = Resolve(name);
             module = _verifier.GetModule(peReader);
 
-            return VerifyAssembly(peReader);
+            return VerifyAssembly(peReader, module);
         }
 
-        private IEnumerable<VerificationResult> VerifyAssembly(PEReader peReader)
+        private IEnumerable<VerificationResult> VerifyAssembly(PEReader peReader, EcmaModule module)
         {
             MetadataReader metadataReader = peReader.GetMetadataReader();
             foreach (var methodHandle in metadataReader.MethodDefinitions)
             {
-                var methodName = metadataReader.GetString(metadataReader.GetMethodDefinition(methodHandle).Name);
+                // get fully qualified method name
+                var methodName = GetQualifiedMethodName(metadataReader, methodHandle);
                 if (ShouldVerifyMethod(methodName))
                 {
+                    _tracker.NotifyMethodProcessing(module, methodHandle, methodName, true);
                     var results = _verifier.Verify(peReader, methodHandle);
                     foreach (var result in results)
                     {
                         yield return result;
                     }
                 }
+                else
+                {
+                    _tracker.NotifyMethodProcessing(module, methodHandle, methodName, false);
+                }
             }
         }
 
+        /// <summary>
+        /// This method returns the fully qualified method name by concatenating assembly, type and method name.
+        /// This method exists to avoid additional assembly resolving, which might be triggered by calling 
+        /// MethodDesc.ToString().
+        /// </summary>
+        private string GetQualifiedMethodName(MetadataReader metadataReader, MethodDefinitionHandle methodHandle)
+        {
+            var methodDef = metadataReader.GetMethodDefinition(methodHandle);
+            var typeDef = metadataReader.GetTypeDefinition(methodDef.GetDeclaringType());
+
+            var methodName = metadataReader.GetString(metadataReader.GetMethodDefinition(methodHandle).Name);
+            var typeName = metadataReader.GetString(typeDef.Name);
+            var namespaceName = metadataReader.GetString(typeDef.Namespace);
+            var assemblyName = metadataReader.GetString(metadataReader.GetAssemblyDefinition().Name);
+
+            StringBuilder builder = new StringBuilder();
+            builder.Append($"[{assemblyName}]");
+            if (string.IsNullOrEmpty(namespaceName) == false)
```

Comment:
I think `if (!string.IsNullOrEmpty...)` should work here. The `== false` makes it look like there is a nullable involved.